src/formatters/deployProgressStatusFormatter.ts

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import * as chalk from 'chalk';
+import { UX } from '@salesforce/command';
+import { Logger } from '@salesforce/core';
+import { getNumber } from '@salesforce/ts-types';
+import { MetadataApiDeployStatus } from '@salesforce/source-deploy-retrieve/lib/src/client/types';
+import { MetadataApiDeploy } from '@salesforce/source-deploy-retrieve';
+import { Duration } from '@salesforce/kit';
+import { ProgressFormatter } from './progressFormatter';
+export class DeployProgressStatusFormatter extends ProgressFormatter {
+  private previousComponents = -1;
+  private previousTests = -1;
+  public constructor(logger: Logger, ux: UX) {
+    super(logger, ux);
+  }
+
+  // This can be used to print the progress of the deployment.
+  public progress(deploy: MetadataApiDeploy): void {
+    deploy.onUpdate((data) => {
+      // Printing status only when number of components or tests gets changed in progress.
+      if (data.numberComponentsDeployed > this.previousComponents || data.numberTestsCompleted > this.previousTests) {
+        this.printDeployStatus(data);
+        this.previousComponents = data.numberComponentsDeployed;
+        this.previousTests = data.numberTestsCompleted;
+      }
+    });
+    deploy.onFinish((data) => {
+      this.printDeployStatus(data.response);
+    });
+    deploy.onError((error: Error) => {
+      throw error;
+    });
+  }
+
+  // Prints Deploying status
+  private printDeployStatus(data: MetadataApiDeployStatus): void {
```

Comment:
This is very similar to `DeployReportResultFormatter`.  I think there is some code sharing that can happen, or maybe this code can even be removed and it can just use that other formatter.  Less code means less to maintain, less chance for bugs, and more consistency in the output between commands.

Let's keep this as is but create a work item to handle possible refactoring.